File: chmod_tool.py
from pathlib import Path
import os
from check_file_permission import check_file_permission

def chmod_file(file_path, mode, config):
    if not check_file_permission(file_path, "write", config):
        raise PermissionError(f"Write access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    os.chmod(file_path, int(mode, 8))
    return f"Changed permissions of {file_path} to {mode}"

File: ts-tools/chmod_tool.ts

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function chmodFile(filePath: string, mode: string, config: any): Promise<string> {
    return new Promise((resolve, reject) => {
        checkFilePermission(filePath, "write", config)
            .then(() => {
                const resolvedPath = path.resolve(filePath);
                fs.chmod(resolvedPath, parseInt(mode, 8), (err) => {
                    if (err) {
                        reject(err);
                    } else {
                        resolve(`Changed permissions of ${filePath} to ${mode}`);
                    }
                });
            })
            .catch(reject);
    });
}
